### PR TITLE
Ag/user api endpoints

### LINKED ISSUE #24: Create CRUD API Endpoints for users
Create API endpoints for CRUD functionality with regard to a user

### PR DESCRIPTION
**Pull request by agallant121**
## Who worked on this
- [ ] Mandy
- [ ] Tanisha
- [x] Alex
- [x] Pierce
## Type of change
- [ ] Style (Non-breaking Style code)
- [x] Testing (Non-breaking Testing code)
- [ ] Bug fix or refactor (change which fixes an issue)
- [x] New feature (change which adds functionality)
## Summary of the change.
🔥 API endpoints now exist for FE to call for CRUD functionality regarding users.
## Link to issue: 
This closes #24 

## Checklist:
- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -2,4 +2,32 @@ class Api::V1::UsersController < ApplicationController
...
     render json: UserSerializer.new(User.all)
   end
+
+  def show
+    render json: UserSerializer.new(User.find(params[:id]))
+  end
+
+  def create
+    user = User.new(user_params)
+    if user.save
```
Comment (agallant121) [inline app/controllers/api/v1/users_controller.rb:12]: Will only render the json response if the user information was filled out correctly.

File: app/controllers/api/v1/users_controller.rb [hunk 2]
```diff
@@ -2,4 +2,32 @@ class Api::V1::UsersController < ApplicationController
...
+      render json: UserSerializer.new(User.create(user_params)), status: 201
+    else
+      user_message = FailedUser.new(user)
```
Comment (agallant121) [inline app/controllers/api/v1/users_controller.rb:15]: Will trigger the failed user poro if user information was not filled in correctly

File: app/controllers/api/v1/users_controller.rb [hunk 3]
```diff
@@ -2,4 +2,32 @@ class Api::V1::UsersController < ApplicationController
...
+      render json: FailedSerializer.new(user_message), status: 401
+    end
+  end
+
+  def update
+    render json: UserSerializer.new(User.update(params[:id], user_params))
```
Comment (agallant121) [inline app/controllers/api/v1/users_controller.rb:21]: Allows user information to be updated.

File: app/controllers/api/v1/users_controller.rb [hunk 4]
```diff
@@ -2,4 +2,32 @@ class Api::V1::UsersController < ApplicationController
...
+  end
+
+  def destroy
```
Comment (agallant121) [inline app/controllers/api/v1/users_controller.rb:24]: Can delete a user from the DB by its id.

File: app/controllers/api/v1/users_controller.rb [hunk 5]
```diff
@@ -2,4 +2,32 @@ class Api::V1::UsersController < ApplicationController
   def index
     render json: UserSerializer.new(User.all)
   end
+
+  def show
+    render json: UserSerializer.new(User.find(params[:id]))
```
Comment (agallant121) [inline app/controllers/api/v1/users_controller.rb:7]: Can show a specific user by their id.

File: config/routes.rb [hunk 6]
```diff
@@ -1,7 +1,7 @@
 Rails.application.routes.draw do
   namespace :api do
     namespace :v1 do
-      resources :users, only: [:index]
+      resources :users, only: [:index, :show, :create, :update, :destroy]
```
Comment (agallant121) [inline config/routes.rb:4]: Added namespaced routes for user crud endpoints.

File: app/controllers/api/v1/users_controller.rb (hunk 2 above)
Comment (palworth) [inline app/controllers/api/v1/users_controller.rb:15]: We decided to create a PORO to store error messages to be serialized

**Comment by palworth** [comment]
Ready for master

**Pull request merged**